core/src/main/resources/org/apache/spark/ui/static/utils.js

```diff
@@ -46,3 +46,31 @@ function formatBytes(bytes, type) {
     var i = Math.floor(Math.log(bytes) / Math.log(k));
     return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
 }
+
+function padZeroes(num) {
+  return ("0" + num).slice(-2);
+}
+
+function formatTimeMillis(timeMillis) {
+  if (timeMillis <= 0) {
+    return "-";
+  } else {
+    var dt = new Date(timeMillis);
+    return dt.getFullYear() + "-" +
+      padZeroes(dt.getMonth() + 1) + "-" +
+      padZeroes(dt.getDate()) + " " +
+      padZeroes(dt.getHours()) + ":" +
+      padZeroes(dt.getMinutes()) + ":" +
+      padZeroes(dt.getSeconds());
+  }
+}
+
+function getTimeZone() {
+  try {
```

Comment:
Some old browser doesn't support this function. so I surround with try catch.

Feature | Chrome | Edge | Firefox | Internet Explorer | Opera | Safari
-- | -- | -- | -- | -- | -- | --
Basic support | 24 | 12 | 29 | 11 | 15 | 10
computed timeZone | 35 | 14 | 53 | No | 30 | 10

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat/resolvedOptions

Reply:
cc @srowen does Spark have a guarantee about which version of which browser to support?

Reply:
Spark does not have a list of supported browsers, from my experience most of us just test new UI changes on whatever browsers we have (In my case the latest Safari, Chrome and Firefox ESR). Though I'd like to think we don't have any IE users, we can't make that assumption while IE 11 is still supported by MS.